task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
+from pyrsistent import pmap
+from pyrsistent import PVector
+from pyrsistent import pvector
+from pyrsistent import v
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+VOLUME_KEYS = set(['mode', 'container_path', 'host_path'])
+
+def valid_volumes(volumes):
+    for vol in volumes:
+        if set(vol.keys()) != VOLUME_KEYS:
+            return (
+                False,
+                'Invalid volume format, must only contain following keys: '
+                '{}, was: {}'.format(VOLUME_KEYS, vol.keys())
+            )
+    return (True, None)
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    uuid = field(type=(str, uuid.UUID), initial=uuid.uuid4)
+    node_name = field(type=str)
+    node_selector = field(type=(str,str))
+    containers = field(type=PVector)
+    restart_policy = field(type=str)
+    # ToDO IAM stuff
+    service_account_name = field(type=str)
```

Comment:
we might wanna circle back to this (or see how PaaSTA sets this) - do we get this for free if we use a kubeconfig file for setting ourselves up or does paasta also put something in here?

Reply:
oh, looks like in paasta we only use this for operators? we can probably remove this then :)

Reply:
Removed.